api_client.go

```diff
@@ -0,0 +1,1031 @@
+                }
+            }
+            if httpClient != nil {
+                cfg.HTTPClient = httpClient
+            }
+        }
+    }
+
+    return c
+}
+
+func (a APIClient) GetStoreId() string {
+    return a.cfg.StoreId
+}
+
+func (a APIClient) SetStoreId(storeId string) {
+    a.cfg.StoreId = storeId
+}
+
+func atoi(in string) (int, error) {
+    return strconv.Atoi(in)
+}
+
+// selectHeaderContentType select a content type from the available list.
+func selectHeaderContentType(contentTypes []string) string {
+    if len(contentTypes) == 0 {
+        return ""
+    }
+    if contains(contentTypes, "application/json") {
+        return "application/json"
+    }
+    return contentTypes[0] // use the first content type specified in 'consumes'
+}
+
+// selectHeaderAccept join all accept types and return
+func selectHeaderAccept(accepts []string) string {
+    if len(accepts) == 0 {
+        return ""
+    }
+
+    if contains(accepts, "application/json") {
+        return "application/json"
+    }
+
+    return strings.Join(accepts, ",")
+}
+
+// contains is a case insensitive match, finding needle in a haystack
+func contains(haystack []string, needle string) bool {
+    for _, a := range haystack {
+        if strings.ToLower(a) == strings.ToLower(needle) {
+            return true
+        }
+    }
+    return false
+}
+
+// Verify optional parameters are of the correct type.
+func typeCheckParameter(obj interface{}, expected string, name string) error {
+    // Make sure there is an object.
+    if obj == nil {
+        return nil
+    }
+
+    // Check the type is as expected.
+    if reflect.TypeOf(obj).String() != expected {
+        return fmt.Errorf("Expected %s to be of type %s but received %s.", name, expected, reflect.TypeOf(obj).String())
+    }
+    return nil
+}
+
+// parameterToString convert interface{} parameters to string, using a delimiter if format is provided.
+func parameterToString(obj interface{}, collectionFormat string) string {
+    var delimiter string
+
+    switch collectionFormat {
+    case "pipes":
+        delimiter = "|"
+    case "ssv":
+        delimiter = " "
+    case "tsv":
+        delimiter = "\t"
+    case "csv":
+        delimiter = ","
+    }
+
+    if reflect.TypeOf(obj).Kind() == reflect.Slice {
+        return strings.Trim(strings.Replace(fmt.Sprint(obj), " ", delimiter, -1), "[]")
+    } else if t, ok := obj.(time.Time); ok {
+        return t.Format(time.RFC3339)
+    }
+
+    return fmt.Sprintf("%v", obj)
+}
+
+// helper for converting interface{} parameters to json strings
+func parameterToJson(obj interface{}) (string, error) {
+    jsonBuf, err := json.Marshal(obj)
+    if err != nil {
+        return "", err
+    }
+    return string(jsonBuf), err
+}
+
+// callAPI do the request.
+func (c *APIClient) callAPI(request *http.Request) (*http.Response, error) {
+    if c.cfg.Debug {
+        dump, err := httputil.DumpRequestOut(request, true)
```

Comment:
maybe move to a helper function?

Reply:
this was auto-generated https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/go/client.mustache#L250